query/src/sql/planner/binder/ddl/database.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+
+use common_ast::ast::CreateDatabaseStmt;
+use common_ast::ast::DatabaseEngine;
+use common_exception::Result;
+use common_meta_app::schema::DatabaseMeta;
+use common_planners::CreateDatabasePlan;
+
+use crate::sql::binder::Binder;
+use crate::sql::plans::Plan;
+
+impl<'a> Binder {
+    pub(in crate::sql::planner::binder) async fn bind_create_database(
+        &self,
+        stmt: &CreateDatabaseStmt<'a>,
+    ) -> Result<Plan> {
+        let catalog = stmt
+            .opt_catalog
+            .as_ref()
+            .map(|catalog| catalog.name.clone())
```

Comment:
We may need a lowercased catalog. cc @leiysky

Reply:
We can leave it as this for now, this is compatible with old planner.